src/GitHub.Api/Installer/GitInstaller.cs

```diff
@@ -11,206 +12,183 @@ class GitInstaller
-                    installationState.GitIsValid = success;
-                    if (success)
-                    {
-                        installationState.GitExecutablePath = path;
-                        installationState.GitInstallationPath = path.Resolve().Parent.Parent;
-                    }
-                };
-                findTask.Then(new FindExecTask("git-lfs", cancellationToken)
-                    .Configure(processManager, dontSetupGit: true))
-                    .Catch(e => true);
-                findTask.OnEnd += (thisTask, path, success, exception) =>
-                {
-                    installationState.GitLfsIsValid = success;
-                    if (success)
-                    {
-                        // we should doublecheck that system git is usable here
-                        installationState.GitLfsExecutablePath = path;
-                        installationState.GitLfsInstallationPath = path.Resolve().Parent.Parent;
-                    }
-                };
-                startTask = findTask.Then(s => installationState);
-            }
-            else
-            {
-                startTask = new FuncTask<GitInstallationState>(cancellationToken, () =>
-                    {
-                        return VerifyPortableGitInstallation();
-                    })
-                { Name = "Git Installation - Verify" };
+                    Logger.Trace("Using git install path from settings: {0}", gitExecutablePath);
+                    state.GitExecutablePath = gitExecutablePath.Value;
+                    state.GitIsValid = true;
+                }
             }
 
-            startTask = startTask.Then(new FuncTask<GitInstallationState, GitInstallationState>(cancellationToken, (success, installState) =>
-                {
-                    if (installState.GitIsValid && installState.GitLfsIsValid)
-                    {
-                        return installState;
-                    }
-
-                    installState = VerifyZipFiles(installState);
-                    installState = GrabZipFromResourcesIfNeeded(installState);
-                    return installState;
-                })
-                { Name = "Git Installation - Validate" }
-            );
-
-            startTask.OnEnd += (thisTask, installState, success, exception) =>
+
+            if (!environment.IsWindows)
             {
-                if (installState.GitIsValid && installState.GitLfsIsValid)
-                {
-                    Logger.Trace("Skipping git installation");
-                    thisTask.Then(installationTask);
-                    return;
-                }
+                return VerifyMacGit(state);
+            }
 
-                var downloadZipTask = DownloadZipsIfNeeded(installState);
-                downloadZipTask.OnEnd += ExtractPortableGit;
-                thisTask.Then(downloadZipTask);
-            };
+            state = VerifyPortableGitInstallation(state);
```

Comment:
If the git path has chosen from settings, this method will overwrite the choice made by settings.
We should definitely attempt to find GitLFS and if both are valid, return `state` before this line of code.

Reply:
Oh yes, I knew I was missing something